narwhal/primary/src/handlers/primary.rs

```diff
@@ -0,0 +1,522 @@
+use crate::metrics::PrimaryMetrics;
+use crate::synchronizer::Synchronizer;
+use config::{SharedCommittee, SharedWorkerCache, WorkerId};
+use crypto::{NetworkPublicKey, PublicKey, Signature};
+use dashmap::DashSet;
+use fastcrypto::hash::Hash;
+use fastcrypto::traits::ToFromBytes;
+use fastcrypto::SignatureService;
+use std::cmp::Reverse;
+use std::collections::{BTreeSet, BinaryHeap};
+use std::sync::Arc;
+use std::time::{Duration, Instant};
+use storage::{CertificateStore, PayloadToken};
+use store::Store;
+use tokio::sync::{oneshot, watch};
+use tracing::{debug, error, info, warn};
+use types::error::{DagError, DagResult};
+use types::metered_channel::Sender;
+use types::{
+    ensure, now, BatchDigest, Certificate, CertificateDigest, FetchCertificatesRequest,
+    FetchCertificatesResponse, GetCertificatesRequest, GetCertificatesResponse, Header,
+    HeaderDigest, PayloadAvailabilityRequest, PayloadAvailabilityResponse, PrimaryMessage,
+    RequestVoteRequest, RequestVoteResponse, Round, Vote, VoteInfo,
+};
+
```

Comment:
The code in this file is pretty much a lift of the existing handler code that used to be under the `primary.rs` file